test/rng_quality.cpp

```diff
@@ -18,7 +18,7 @@ template<typename T>
 class RandomEngine : public ::testing::Test {
    public:
     virtual void SetUp() {
-        //Ensure all unlocked buffers are freed
+        // Ensure all unlocked buffers are freed
         deviceGC();
```

Comment:
If you want to unlock all buffers, you need to add af::sync() before the deviceGC().
Without the sync, the buffers which are still in use by executing kernels will not be freed.

Reply:
That is not the case with `deviceGC` in this location. When test finishes, all arrays involved are already evaluated because they need to be brought to host for comparison. So, they will be marked as unused by the time this line hits before the next test starts.

Reply:
Additionally all allocation and free function are device synchronization events.